src/aks-preview/azext_aks_preview/_consts.py

```diff
@@ -180,6 +180,16 @@
 # only used as the key of the corresponding description, not to map to the key name in addonProfiles,
 # since its configuration is actually stored in a separate ingress profile
 CONST_WEB_APPLICATION_ROUTING_KEY_NAME = "ingress/webApplicationRouting"
+CONST_APP_ROUTING_ANNOTATION_CONTROLLED_NGINX = "annotationControlled"
+CONST_APP_ROUTING_EXTERNAL_NGINX = "external"
+CONST_APP_ROUTING_INTERNAL_NGINX = "internal"
+CONST_APP_ROUTING_NONE_NGINX = "none"
+APP_ROUTING_NGINX_TO_API = {
+    CONST_APP_ROUTING_ANNOTATION_CONTROLLED_NGINX: "AnnotationControlled",
+    CONST_APP_ROUTING_EXTERNAL_NGINX: "External",
+    CONST_APP_ROUTING_INTERNAL_NGINX: "Internal",
+    CONST_APP_ROUTING_NONE_NGINX: "None"
+}
```

Comment:
If I understand this correctly, the mapping here is to ensure that the corresponding fields in the request body have the first letter capitalized? If so, you don't need to do this mapping. Just change the constant to capitalize the first letter. The argument parser will handle the case conversion properly (e.g., with current settings, user input "annotationControlled", "AnnotationControlled", "ANNOTATIONCONTROLLED" will all be parsed as "annotationControlled")

Reply:
got it, thanks for the pointer!